README.md

```diff
@@ -324,6 +324,41 @@ The following SDK library releases are available on [vcpkg](https://github.com/m
 
 > NOTE: In case of getting linker errors when consuming the SDK on Windows, make sure that [vcpkg triplet](https://vcpkg.readthedocs.io/en/latest/users/triplets/) being consumed matches the [CRT link flags](https://docs.microsoft.com/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160) being set for your app or library build. See also `MSVC_USE_STATIC_CRT` build flag.
 
+## OpenSSL Version
+
+Several packages within the Azure SDK for C++ use the OpenSSL library. By default, the Azure SDK will use whatever the most recent version of OpenSSL is within the VCPKG repository.
+
+If you need to use a specific version of OpenSSL, you can use the vcpkg custom ports feature to specify the version of OpenSSL to use. 
+For example, if you want to use OpenSSL 1.1.1, you should create a folder named `vcpkg-custom-ports` next to to your vcpkg.json file.
+
+Navigate to your clone of the vcpkg vcpkg repo and execute "git checkout 3b3bd424827a1f7f4813216f6b32b6c61e386b2e" - this will reset your repo to the last version of OpenSSL 1.1.1

```

Comment:
Does this pin the version to `1.1.1n`? Is there any way to upgrade to newer versions like `1.1.1o` or `1.1.1r`?

Is the FIPS certification the problem or one of problems we're trying to solve in this PR? If so, pinning to an older version doesn't seem to meet the requirement.

Reply:
Yes it does, because that is the most recent FIPS certified version of OpenSSL (and more importantly the last version supported on vcpkg).

There are two problems we're trying to solve with this PR:
1. Customers who are running on OS's which don't support OpenSSL 3 can't use OpenSSL 3.
2. Customers who require FIPS certification cannot use the Azure SDK.

It *is* possible to switch to OpenSSL 1.1.1q (which is the most recent version published on OpenSSL.org) - I did it while testing this PR. It's a bit convoluted and we don't want to commit to long term support for those versions of OpenSSL so I didn't include instructions on how to do this.

Reply:
In the cryptography security guidance, it says

- Do not hardcode to OpenSSL 1.1.1.
- OpenSSL 1.1.1* is allowed and **must** use a recent release version that includes patches for all known vulnerabilities and **must stay up to date** regarding known vulnerability fixes.

I think technically we hardcode the version in this PR. In addition, how do we upgrade to new versions if there was a new vulnerability found in `1.1.1n` someday?